http.go

```diff
@@ -68,15 +69,30 @@ func (z *Zoho) HTTPRequest(endpoint *Endpoint) (err error) {
 	)
 
 	// Has a body, likely a CRUD operation (still possibly JSONString)
-	if endpoint.RequestBody != nil {
+	if endpoint.RequestBody != nil{
+
+		// TODO: Decide whether JSON.marshal is required or not for
+		// Bookings or CRM
+
 		// JSON Marshal the body
-		marshalledBody, err := json.Marshal(endpoint.RequestBody)
-		if err != nil {
-			return fmt.Errorf("Failed to create json from request body")
+		switch v := endpoint.RequestBody.(type) {
+			case map[string]string:
+				body := url.Values{}
+				for k, val := range v {
+					body.Add(k,val)
+				}
+
+				reqBody = strings.NewReader(body.Encode())
+				contentType = "application/x-www-form-urlencoded; charset=UTF-8"
+			default:
+				marshalledBody, err := json.Marshal(v)
+				if err != nil {
+					return fmt.Errorf("Failed to create json from request body")
+				}
+
+				reqBody = bytes.NewBuffer(marshalledBody)
+				contentType = "application/x-www-form-urlencoded; charset=UTF-8"
 		}
```

Comment:
This is a little fragile I expect. We will discuss how to structure this once we decide the best way to provide the proper formatting to the `requestBody`.

Reply:
Sure, the primary motivation to these changes were to accomodate different variable types which could come up in `requestBody` parameter. At the moment, type switching made an apt sense for me. I am open to any other ideas regarding the structure used for different `requestBody` formats.